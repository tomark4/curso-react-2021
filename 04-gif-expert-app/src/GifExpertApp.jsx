import AddCategory from "./components/AddCategory";
import {useState} from 'react'
import GifGrid from "./components/GifGrid";

const GifExpertApp = () => {
  const [categories, setCategories] = useState([
    { id: 1, name: "Dragon ball" },
  ]);

  const handleAdd = (name) => {
    const lastItem = categories[categories.length - 1];
    setCategories([
      ...categories,
      { id: lastItem.id + 1, name },
    ]);
  };

  return (
    <>
      <h2>Soy la gif expert app</h2>
      <AddCategory onAdd={ handleAdd }/>
      <hr />
      {categories.map( ({id, name}) => (
          <GifGrid name={name} key={id} />
      ))}
    </>
  );
};

export default GifExpertApp;
