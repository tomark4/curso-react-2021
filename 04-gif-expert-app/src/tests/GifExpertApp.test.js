import { shallow } from "enzyme"
import GifExpertApp from "../GifExpertApp"

describe('Pruebas en GifExpertApp Component', () => {

  test('should be return component', () => {
    const wrapper = shallow( <GifExpertApp />)
    expect( wrapper ).toMatchSnapshot();
  });

  test('debe de mostrar 1 elemento en GifGrid al cargar', () => {
    const categories = [
      { id: 1, name: "Dragon ball" },
    ]

    const wrapper = shallow(<GifExpertApp /> )
    expect( wrapper ).toMatchSnapshot()

    expect( wrapper.find('GifGrid').length ).toBe(categories.length)
  })


});
