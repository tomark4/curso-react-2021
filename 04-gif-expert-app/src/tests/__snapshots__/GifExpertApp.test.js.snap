// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp Component debe de mostrar 1 elemento en GifGrid al cargar 1`] = `
<Fragment>
  <h2>
    Soy la gif expert app
  </h2>
  <AddCategory
    onAdd={[Function]}
  />
  <hr />
  <GifGrid
    key="1"
    name="Dragon ball"
  />
</Fragment>
`;

exports[`Pruebas en GifExpertApp Component should be return component 1`] = `
<Fragment>
  <h2>
    Soy la gif expert app
  </h2>
  <AddCategory
    onAdd={[Function]}
  />
  <hr />
  <GifGrid
    key="1"
    name="Dragon ball"
  />
</Fragment>
`;
